frontend/appflowy_tauri/src/appflowy_app/components/layout/NavigationPanel/FolderItem.hooks.ts

```diff
@@ -30,57 +47,81 @@ export const useFolderEvents = (folder: IFolder) => {
     setShowRenamePopup(true);
   };
 
-  const changeFolderTitle = (newTitle: string) => {
+  const changeFolderTitle = async (newTitle: string) => {
+    await appBackendService.update({ name: newTitle });
     appDispatch(foldersActions.renameFolder({ id: folder.id, newTitle }));
   };
 
   const closeRenamePopup = () => {
     setShowRenamePopup(false);
   };
 
-  const deleteFolder = () => {
+  const deleteFolder = async () => {
     closePopup();
+    await appBackendService.delete();
     appDispatch(foldersActions.deleteFolder({ id: folder.id }));
   };
 
-  const duplicateFolder = () => {
+  const duplicateFolder = async () => {
     closePopup();
-    appDispatch(foldersActions.addFolder({ id: nanoid(8), title: folder.title }));
+
+    const newApp = await workspaceBackendService.createApp({
+      name: folder.title,
+    });
+    appDispatch(foldersActions.addFolder({ id: newApp.id, title: folder.title }));
   };
 
   const closePopup = () => {
     setShowFolderOptions(false);
     setShowNewPageOptions(false);
   };
 
-  const onAddNewDocumentPage = () => {
+  const onAddNewDocumentPage = async () => {
     closePopup();
+    const newView = await appBackendService.createView({
```

Comment:
Are there any global error page? If we have, then the errors can be shown there. Every call to the backend service, it comes with a result. It might success or failed. I think we should display the errors first.

Reply:
I've created Error Handler page Error modal and corresponding redux functions to show error messages. Now any error should be properly handled.